src/api/gamestate.cc

```diff
@@ -1,13 +1,48 @@
 #include "gamestate.h"
 #include <vector>
 #include <memory>
+#include <cstdlib>
+#include <unordered_map>
+#include <iostream>
+#include <cstring>
+#include <cerrno>
+#include <new>
 #include "../statefunctions/statecontroller.h"
```

Comment:
Avoid relative includes, you should be able to reference it from the src root. If not we should amend the cmake for this unit.